File: backend/internal/models/known_host.go
package models

type KnownHost struct {
	ID          string    `json:"id"`
	Hostname    string    `json:"hostname"`
	Port        int       `json:"port"`
	Fingerprint string    `json:"fingerprint"`
	PublicKey   string    `json:"publicKey"`
}

type HostKeyVerification struct {
	Status      string `json:"status"` // "new", "known", "changed"
	Hostname    string `json:"hostname"`
	Port        int    `json:"port"`
	Fingerprint string `json:"fingerprint"`
	KeyType     string `json:"keyType"`
	OldFingerprint string `json:"oldFingerprint,omitempty"`
}

File: mobile/src/types/knownHost.ts
export interface KnownHost {
  id: string
  hostname: string
  port: number
  fingerprint: string
  publicKey: string
}

export interface HostKeyVerification {
  status: 'new' | 'known' | 'changed'
  hostname: string
  port: number
  fingerprint: string
  keyType: string
  oldFingerprint?: string
}
